pkg/keyspace/tso_keyspace_group.go

```diff
@@ -388,6 +388,20 @@ func (m *GroupManager) getKeyspaceConfigByKindLocked(userKind endpoint.UserKind)
 	return config, nil
 }
 
+// GetGroupByKeyspaceID returns the keyspace group ID for the given keyspace ID.
+func (m *GroupManager) GetGroupByKeyspaceID(id uint32) (uint32, error) {
+	m.RLock()
+	defer m.RUnlock()
+	for _, groups := range m.groups {
```

Comment:
Could we maintain a `keyspaceLookupTable` as we did in `pkg/tso/keyspace_group_manager.go`? It would be more efficient than the pure iteration to check which a keyspace ID belongs to.

Reply:
Maybe, but I don't think it's necessary at the moment. We're only accessing this via pd-ctl right now, and it's a low-frequency operation.

Reply:
cc @JmPotato @rleungx

Reply:
After some discussion, we decided to not change it until we encounter performance issues.